python/ray/dag/compiled_dag_node.py

```diff
@@ -681,13 +684,10 @@ def __init__(
         # Mapping from the actor handle to the node ID that the actor is on.
         self.actor_to_node_id: Dict["ray.actor.ActorHandle", str] = {}
 
-        # This is set to true when type hint of `transport="nccl"` is used.
-        self._use_default_nccl_group = False
-        # This is set to the specified custom nccl group
-        # if there exists a type hint of `transport=nccl_group`.
-        self._custom_nccl_group_p2p: Optional[GPUCommunicator] = None
-        # The NCCL group ID for P2P send/recv operations.
-        self._nccl_group_id_p2p: Optional[str] = None
+        # The `custom_nccl_group` specified in `experimental_compile`.
+        self._custom_nccl_group_p2p: Optional[GPUCommunicator] = custom_nccl_group
```

Comment:
```suggestion
        self._default_nccl_group_p2p: Optional[GPUCommunicator] = custom_nccl_group
```

Is this ever different from the group ID in the next line?

Reply:
<details><summary>There were 2 cases where they may be different:</summary>
<ul><li>If a default NCCL group is specified in `experimental_compile`, but it is never used in the DAG (e.g., there is no P2P NCCL send/recvs in the DAG), the `default_nccl_group_id_p2p` will be None.</li>
<li>If no default NCCL group is specified in `experimental_compile`, but there are some NCCL send/recv operations in the DAG whose `transport="nccl"`, either a NCCL group specified somewhere else in the DAG is reused for these P2P operations, or a new `_NcclGroup` is created. In this case, `default_nccl_group_id_p2p` is set to the id of that (reused/created) NCCL group.</li></ul>
</details>

To make it simpler, I've set it to the `default_nccl_group_p2p`'s ID, if one was provided and used, and None otherwise.

In the case where a default NCCL group is provided but never used (and thus never initialized) in the DAG, the ID is still set to None.